aws-import/awsimport.py

```diff
@@ -0,0 +1,257 @@
+# There are also AWS docs in markdown format at https://github.com/awsdocs but more complex
+# metric descriptions had formatting errors and sometimes linked out to the HTML docs themselves.
+
+# https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html
+import argparse
+import re
+from abc import abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from textwrap import dedent
+from typing import Set, List, Mapping
+
+import requests
+from bs4 import BeautifulSoup
+from tabulate import tabulate
+
+RE_WHITESPACE = re.compile(r"\s+")
+RE_REMOVE_PARENS = re.compile(r"^\(.*\)")
+
+
+def _clean_whitespace(s):
+    """Replaces consecutive whitespace with a single space anywhere inside the string."""
+    return RE_WHITESPACE.sub(" ", s).strip()
+
+
+def _remove_parens(s):
+    """Removes all text inside parenthesis at the beginning of the string"""
+    return RE_REMOVE_PARENS.sub("", s).strip()
+
+def _replace_brackets(s):
+    """Replace brackets with parenthesis because it breaks the UI"""
+    # It probably thinks the brackets are markdown or something.
+    return s.replace("[", "(").replace("]", ")").strip()
+
+
+@dataclass
+class Metric:
+    name: str
+    brief: str
+
+
+class Extractor:
+    @abstractmethod
+    def extract(self, html) -> List[Metric]:
+        pass
+
+
+def get_description(desc):
+    clean, *rest = _replace_brackets(_remove_parens(_clean_whitespace(desc.text))).split(".")
```

Comment:
Would be useful to have the full text in the description instead of throwing away.  Also I would use the nltk to parse sentences instead of assuming first `.` is the end of the first sentence.

Reply:
Unfortunately they sometimes have html formatting with bullet points etc. for some metrics. We can improve it later but I'm also not clear where the long description would even be shown to users.